src/OrchardCore.Modules/OrchardCore.Security/Views/SecurityHeadersSettings.Edit.cshtml

```diff
@@ -0,0 +1,329 @@
+                name = T["Push"];
+                break;
+            case PermissionsPolicyValue.SpeakerSelection:
+                name = T["Speaker Selection"];
+                break;
+            case PermissionsPolicyValue.ScreenWakeLock:
+                name = T["Screen Wake Lock"];
+                break;
+            case PermissionsPolicyValue.SyncXhr:
+                name = T["Synchronous XML Http Request"];
+                break;
+            case PermissionsPolicyValue.UnoptimizedImages:
+                name = T["Unoptimized Images"];
+                break;
+            case PermissionsPolicyValue.UnsizedMedia:
+                name = T["Unsized Media"];
+                break;
+            case PermissionsPolicyValue.Usb:
+                name = T["USB"];
+                break;
+            case PermissionsPolicyValue.WebShare:
+                name = T["Web Share"];
+                break;
+            case PermissionsPolicyValue.WebXR:
+                name = T["Web XR"];
+                break;
+        }
+
+        return name;
+    }
+}
+
+<p class="alert alert-warning">
+    @T["The current tenant will be reloaded when the settings are saved."]
+</p>
+
+@if (Model.FromConfiguration)
+{
+    <p class="alert alert-info">
+        @T["The current settings are coming from configuration sources, saving the settings will affect the AdminSettings not the configuration."]
+    </p>
+}
+
+<div class="form-group">
+    <label asp-for="ContentSecurityPolicy">@T["Content Security Policy"]</label>
+    <div class="form-group" asp-validation-class-for="EnableSandbox">
+        <div class="custom-control custom-checkbox">
+            <input type="checkbox" class="custom-control-input" asp-for="EnableSandbox" />
+            <label class="custom-control-label" asp-for="EnableSandbox">@T["Enable Sandbox"]</label>
+        </div>
+    </div>
+    <div class="form-group" asp-validation-class-for="UpgradeInsecureRequests">
+        <div class="custom-control custom-checkbox">
+            <input type="checkbox" class="custom-control-input" asp-for="UpgradeInsecureRequests" />
+            <label class="custom-control-label" asp-for="UpgradeInsecureRequests">@T["Upgrade Insecure Requests"]</label>
+        </div>
+    </div>
+    <table class="table table-striped table-hover">
+        <thead>
+            <tr>
+                <th>@T["Name"]</th>
+                <th></th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var key in Model.ContentSecurityPolicy.Keys)
+            {
+                var value = Model.ContentSecurityPolicy[key];
+
+                <tr>
+                    <td>@GetContentSecurityPolicyName(key)</td>
+                    <td>
+                        <div class="form-group mb-0">
+                            <input id="@Html.IdFor(m => m.ContentSecurityPolicy[key]))" name="@Html.NameFor(m => m.ContentSecurityPolicy[key]))" type="hidden" value="@value"/>
```

Comment:
Now that we only have this input, it needs `class="form-control"`, there is an extra `)`, should not be `type="hidden"` anymore, and as I remember good to have a space at the end just before the closing `/>`

```suggestion
                            <input class="form-control" id="@Html.IdFor(m => m.ContentSecurityPolicy[key])" name="@Html.NameFor(m => m.ContentSecurityPolicy[key])" value="@value" />
```

One simple way is to just copy paste this file from my branch